src/resultset/resultset.h

```diff
@@ -16,7 +16,7 @@
 #define RESULTSET_OK 1
 #define RESULTSET_FULL 0
 
-typedef struct {
+typedef struct ResultSet {
```

Comment:
I'm not sure we need to include the name of the struct at the beginning and at the end, one of them should be removed.

Reply:
need otherwise we can't declare typedef later